front-end/src/components/main-page/TranscriptUploadModal.jsx

```diff
@@ -50,7 +50,18 @@ class TranscriptUploadModal extends React.Component {
   validateData(data) {
     try {
       const parsedData = JSON.parse(data);
-      return parsedData.data;
+      if (parsedData.data){
+        for (var i = 0; i < parsedData.data.length; i++) {
+          const intent = parsedData.data[i]
+          if (!(intent.hasOwnProperty("project_id") && intent.hasOwnProperty("trace_type") &&
+              intent.hasOwnProperty("trace_payload"))){
```

Comment:
Maybe either move this into a helper function, or comment which properties it is checking for. Also we can remove hasOwnProperty and replace it with:
```suggestion
          if (!intent.project_id || !intent.trace_type || !intent.trace_payload){
```
Since null <=> not in this case